pkg/logs/tailer.go

```diff
@@ -0,0 +1,199 @@
+package logs
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/logrusorgru/aurora"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/stripe/stripe-cli/pkg/ansi"
+	"github.com/stripe/stripe-cli/pkg/stripeauth"
+	"github.com/stripe/stripe-cli/pkg/websocket"
+)
+
+const outputFormatJSON = "json"
+
+// LogFilters contains all of the potential user-provided filters for log tailing
+type LogFilters struct {
+	FilterIPAddress      string `json:"filter_ip_address,omitempty"`
+	FilterHTTPMethod     string `json:"filter_http_method,omitempty"`
+	FilterRequestPath    string `json:"filter_request_path,omitempty"`
+	FilterSource         string `json:"filter_source,omitempty"`
+	FilterStatusCode     string `json:"filter_status_code,omitempty"`
+	FilterStatusCodeType string `json:"filter_status_code_type,omitempty"`
+}
+
+// Config provides the cfguration of a Proxy
+type Config struct {
+	APIBaseURL string
+
+	// DeviceName is the name of the device sent to Stripe to help identify the device
+	DeviceName string
+
+	// Filters for API request logs
+	Filters *LogFilters
+
+	// Key is the API key used to authenticate with Stripe
+	Key string
+
+	// Info, error, etc. logger. Unrelated to API request logs.
+	Log *log.Logger
+
+	// Force use of unencrypted ws:// protocol instead of wss://
+	NoWSS bool
+
+	// Output format for request logs
+	OutputFormat string
+
+	// WebSocketFeature is the feature specified for the websocket connection
+	WebSocketFeature string
+}
+
+// Tailer is the main interface for running the log tailing session
+type Tailer struct {
```

Comment:
Same with this, I think the webhook one is identical.

Reply:
This is similar as well, but does not contain `endpointClients  []*endpoint.Client`, which the webhook one does contain. Each also has their own specific config. It is similar code, but it's also not much, and for similar reasons to the above case, I think it's safer and cleaner to leave them separate.